torch/cuda/__init__.py

```diff
@@ -165,6 +157,31 @@ def is_bf16_supported():
     # Finally try to create a bfloat16 device.
     return _check_bf16_tensor_supported(device)
 
+@lru_cache(maxsize=256)
+def _is_bf16_compute_supported(device):
+    if not _is_bf16_supported(device):
+        return False
+    try:
+        a = torch.zeros(1, 1, 1, 1, device=torch.device('cuda', device))
+        torch.nn.functional.conv2d(a, a, a.reshape(1), 1, 1, 1, 1)
+        torch.matmul(a, a)
```

Comment:
If we want to properly check the full support, we would need to add `torch.compile` here as OAI/Triton fails for `bfloat16` ops on e.g. Volta.
Shouldn't we just restore the previously used `sm_80+` check instead?